httpbp/middlewares.go

```diff
@@ -362,6 +369,12 @@ func statusCodeFamily(code int) string {
 	return families[family]
 }
 
+// isSuccessStatusCode takes an http status code and returns true if its in the
+// success status code family, i.e "2xx".
+func isSuccessStatusCode(code int) bool {
+	return code/100 == 2
+}
```

Comment:
I think by definition any http code < 400 should be considered as "success" (e.g. a 3xx redirect is not a failure, nor is 101 switch protocol).

@kylelemons ?

Reply:
i agree. at a minimum the baseplate.spec says 2xx or 3xx  so i will add that

Reply:
If you get a 3xx from the http client (i.e. not from RoundTrip) it should be considered a failure because it is not a terminal status code.  Same with 1xx, I believe; 101 switching protocols is nonterminal.

Reply:
Nevermind, this is used in a RoundTripper, so considering `code >= 100 && code < 400` successful sounds fine to me

Reply:
1. On server side 3xx certainly should be considered success (currently this function is shared between server code and client code), I'm less familiar with 1xx
2. Although go's default client will follow redirections (with 10 max), you can configure the client to not follow redirections. So if the client is configured to follow redirections, you should not get 3xx response codes (if the redirect max is hit you get an error instead, I believe, not 100% sure); but if the client is configured to not follow redirections, the client can get 3xx responses and those should be considered success.